python/cudf/cudf/tests/test_concat.py

```diff
@@ -993,9 +1005,29 @@ def test_concat_join_empty_dataframes(
         other_gd, ignore_index=ignore_index, axis=axis, join=join, sort=sort
     )
     if expected.shape != df.shape:
-        for key, col in actual[actual.columns].iteritems():
-            if is_categorical_dtype(col.dtype):
-                expected[key] = expected[key].fillna("-1")
-                actual[key] = col.astype("str").fillna("-1")
-        assert_eq(expected.fillna(-1), actual.fillna(-1), check_dtype=False)
-    assert_eq(expected, actual, check_index_type=False if gdf.empty else True)
+        if actual.empty and not any(expected.columns == actual.columns):
+            # when join='inner' and all input are empty dfs
+            # pandas returns range column, we return index column
+            assert True
```

Comment:
Instead of asserting True, we should be able to pass `check_column_type=False` to have it ignore `RangeIndex` vs `Index` for the column names.

Reply:
Awesome, sounds good!